src/main/scala/io/iohk/ethereum/network/PeerMessageBusActor.scala

```diff
@@ -27,57 +27,64 @@ object PeerMessageBusActor {
 
   case class MessageFromPeer(message: Message, peerId: PeerId)
 
-  case class Subscription(subscriber: ActorRef, classifier: MessageClassifier)
-
   class PeerMessageBus extends ActorEventBus {
 
     override type Event = MessageFromPeer
     override type Classifier = MessageClassifier
 
-    private var subscriptions: Seq[Subscription] = Nil
+    //FIXME Remove this var
+    private var subscriptions: Map[(Subscriber, PeerSelector), Set[Int]] = Map.empty
```

Comment:
We can but I rather to do it on another PR as it was not introduced in this fix. 

@adamsmo do you think is worthy to do it now? :innocent: